services/httpd/response_writer.go

```diff
@@ -147,7 +148,29 @@ func (w *csvFormatter) WriteResponse(resp Response) (n int, err error) {
 			}
 		}
 
-		for _, row := range result.Series {
+		for i, row := range result.Series {
+			if i > 0 && !reflect.DeepEqual(result.Series[i-1].Columns, row.Columns) {
```

Comment:
Since these are both string slices, can you make a helper method to compare them instead of using reflect?

Reply:
Sure.